### DISCUSSION TITLE
Evaluating Alpine syntax inside x-html that is within a template block

### DISCUSSION THREAD

**Discussion by fredcarle** [alpinejs/alpine#1818] [7. Bugs (archived)]
Hey @calebporzio! I'm trying this with version 3.2.0 and 3.2.2 and in either case the Alpine syntax inside the x-html is not evaluated when it's within a template block.

``` html
<ul x-data="{items: [...], x: 12}">
  <li x-html="`<h2 x-text=x></h2>`"></li> <!-- This works  -->
  <template x-for="(item,i) in items">
     <li x-html="`<h2 x-text=item></h2>`"></li> <!-- This does not-->
  </template>
</ul>
```

**Comment by tvdr** [comment]
i submitted a PR (#1819) which fixes this

also there is a workaround for this:

```javascript
<li x-init="$nextTick(()=>{$el.innerHTML = `<h2 x-text=item></h2>`})"></li> 
```

**Reply by fredcarle** [reply]
Thanks a lot for that workaround. It works just fine. Hopefully the fix will be integrated into the next release.
